### ISSUE TITLE
[Bug]: prepare problem 

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#5795]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): d1518041fd0c40c8315a7c9148be46dc2e052f9c
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

1. an error occurred when handling prepare query with `JSON` type.
2. in the app side, this request doesn't be responded to until mo server crashed.

```
{"level":"INFO","time":"2022/10/20 11:46:12.378716 +0800","caller":"frontend/util.go:463","msg":"query trace status","connection_id":1191,"statement":"prepare __mo_stmt_id_63 from select enum_id, name, description, profile, deleted_at, created_at, updated_at from system_profile where enum_id = ?","status":"success"}
{"level":"INFO","time":"2022/10/20 11:46:12.378755 +0800","caller":"frontend/routine.go:170","msg":"connection id 1191 , the time of handling the request 3.793726ms"}
{"level":"INFO","time":"2022/10/20 11:46:12.379378 +0800","caller":"frontend/mysql_cmd_executor.go:2630","msg":"cmd 23"}
{"level":"INFO","time":"2022/10/20 11:46:12.379455 +0800","caller":"frontend/mysql_cmd_executor.go:2790","msg":"query trace","connection_id":1191,"query":"execute __mo_stmt_id_63 using @__mo_stmt_var_0","vars":"105"}
{"level":"INFO","time":"2022/10/20 11:46:12.379623 +0800","caller":"frontend/util.go:463","msg":"query trace status","connection_id":4294967295,"statement":"start transaction","status":"success"}
{"level":"INFO","time":"2022/10/20 11:46:12.379744 +0800","caller":"frontend/session.go:1301","msg":"dbName mo_catalog tableNames [mo_database mo_tables mo_columns %!%mo_increment_columns mo_user mo_account mo_role mo_user_grant mo_role_grant mo_role_privs]"}
{"level":"INFO","time":"2022/10/20 11:46:12.379814 +0800","caller":"frontend/session.go:1301","msg":"dbName mo_catalog tableNames [mo_database mo_tables mo_columns %!%mo_increment_columns mo_user mo_account mo_role mo_user_grant mo_role_grant mo_role_privs]"}
{"level":"INFO","time":"2022/10/20 11:46:12.379941 +0800","caller":"frontend/mysql_cmd_executor.go:2314","msg":"time of Exec.Build : 242.253µs"}
{"level":"INFO","time":"2022/10/20 11:46:12.380136 +0800","caller":"compile/scope.go:43","msg":"\t:2 filter(typ:<id:10 size:1 > f:<func:<obj:73014444032 > args:<typ:<id:10 size:1 > f:<func:<obj:7 obj_name:\"=\" > args:<typ:<id:23 size:8 > f:<func:<obj:90194313262 obj_name:\"cast\" > args:<typ:<id:28 width:64 > col:<col_pos:2 name:\"rp.obj_id\" > > args:<t:<typ:<id:23 size:8 > > > > > args:<typ:<id:23 size:8 > c:<ival:0 > > > > args:<typ:<id:10 size:1 > f:<func:<obj:73014444032 > args:<typ:<id:10 size:1 > f:<func:<obj:12 obj_name:\"=\" > args:<typ:<id:61 width:16 > col:<col_pos:1 name:\"rp.obj_type\" > > args:<typ:<id:61 width:5 size:4 > c:<sval:\"table\" > > > > args:<typ:<id:10 size:1 > f:<func:<obj:73014444032 > args:<typ:<id:10 size:1 > f:<func:<obj:7 obj_name:\"=\" > args:<typ:<id:23 size:8 > f:<func:<obj:90194313258 obj_name:\"cast\" > args:<typ:<id:22 width:32 > col:<name:\"rp.role_id\" > > args:<t:<typ:<id:23 size:8 > > > > > args:<typ:<id:23 size:8 > c:<ival:0 > > > > args:<typ:<id:10 size:1 > f:<func:<obj:73014444032 > args:<typ:<id:10 size:1 > f:<func:<obj:7 obj_name:\"=\" > args:<typ:<id:23 size:8 > f:<func:<obj:90194313258 obj_name:\"cast\" > args:<typ:<id:22 width:32 > col:<col_pos:3 name:\"rp.privilege_id\" > > args:<t:<typ:<id:23 size:8 > > > > > args:<typ:<id:23 size:8 > c:<ival:30 > > > > args:<typ:<id:10 size:1 > f:<func:<obj:12 obj_name:\"=\" > args:<typ:<id:61 width:100 > col:<col_pos:4 name:\"rp.privilege_level\" > > args:<typ:<id:61 width:3 size:4 > c:<sval:\"*.*\" > > > > > > > > > > > ) -> projection(typ:<id:22 width:32 > col:<col_pos:3 name:\"rp.privilege_id\" > ,typ:<id:10 > col:<col_pos:5 name:\"rp.with_grant_option\" > ) -> projection(typ:<id:22 width:32 > col:<name:\"rp.privilege_id\" > ,typ:<id:10 > col:<col_pos:1 name:\"rp.with_grant_option\" > ) -> pipe connector"}
{"level":"INFO","time":"2022/10/20 11:46:12.380167 +0800","caller":"compile/scope.go:43","msg":":0  union all  -> sql output"}
{"level":"INFO","time":"2022/10/20 11:46:12.380730 +0800","caller":"frontend/mysql_cmd_executor.go:2405","msg":"time of Exec.Run : 738.098µs"}
{"level":"INFO","time":"2022/10/20 11:46:12.380820 +0800","caller":"frontend/util.go:463","msg":"query trace status","connection_id":4294967295,"statement":"select rp.privilege_id, rp.with_grant_option from mo_catalog.mo_role_privs as rp where rp.obj_id = 0 and rp.obj_type = \"table\" and rp.role_id = 0 and rp.privilege_id = 30 and rp.privilege_level = \"*.*\"","status":"success"}
{"level":"INFO","time":"2022/10/20 11:46:12.380881 +0800","caller":"frontend/util.go:463","msg":"query trace status","connection_id":4294967295,"statement":"commit","status":"success"}
{"level":"INFO","time":"2022/10/20 11:46:12.380896 +0800","caller":"frontend/mysql_cmd_executor.go:2812","msg":"----close mce"}
{"level":"INFO","time":"2022/10/20 11:46:12.380962 +0800","caller":"frontend/mysql_cmd_executor.go:2314","msg":"time of Exec.Build : 1.451891ms"}
{"level":"INFO","time":"2022/10/20 11:46:12.381314 +0800","caller":"compile/scope.go:43","msg":"\t:2 filter(typ:<id:10 size:1 > f:<func:<obj:7 obj_name:\"=\" > args:<typ:<id:23 size:8 > f:<func:<obj:90194313258 obj_name:\"cast\" > args:<typ:<id:22 size:4 > f:<func:<obj:90194313218 obj_name:\"cast\" > args:<typ:<id:22 width:32 > col:<> > args:<typ:<> t:<typ:<id:22 size:4 > > > > > args:<t:<typ:<id:23 size:8 > > > > > args:<typ:<id:23 size:8 > c:<ival:105 > > > ) -> projection(typ:<id:22 width:32 > col:<> ,typ:<id:61 width:255 > col:<col_pos:1 > ,typ:<id:61 width:255 > col:<col_pos:2 > ,typ:<id:62 > col:<col_pos:3 > ,typ:<id:53 > col:<col_pos:4 > ,typ:<id:53 > col:<col_pos:5 > ,typ:<id:53 > col:<col_pos:6 > ) -> projection(typ:<id:22 width:32 > col:<> ,typ:<id:61 width:255 > col:<col_pos:1 > ,typ:<id:61 width:255 > col:<col_pos:2 > ,typ:<id:62 > col:<col_pos:3 > ,typ:<id:53 > col:<col_pos:4 > ,typ:<id:53 > col:<col_pos:5 > ,typ:<id:53 > col:<col_pos:6 > ) -> pipe connector"}
{"level":"INFO","time":"2022/10/20 11:46:12.381342 +0800","caller":"compile/scope.go:43","msg":":0  union all  -> sql output"}
{"level":"INFO","time":"2022/10/20 11:46:12.381730 +0800","caller":"containers/utils.go:227","msg":"sie : 5"}
{"level":"INFO","time":"2022/10/20 11:46:12.381851 +0800","caller":"containers/utils.go:227","msg":"sie : 1"}
{"level":"INFO","time":"2022/10/20 11:46:12.382058 +0800","caller":"frontend/mysql_cmd_executor.go:439","msg":"goid 616951 \n"}
{"level":"INFO","time":"2022/10/20 11:46:12.382159 +0800","caller":"frontend/mysql_protocol.go:1812","msg":"error: internal error: type is not supported in binary text result row","span":{}}
{"level":"ERROR","time":"2022/10/20 11:46:12.382242 +0800","caller":"frontend/mysql_cmd_executor.go:305","msg":"flush error internal error: type is not supported in binary text result row \n","stacktrace":"github.com/matrixorigin/matrixone/pkg/frontend.(*outputQueue).flush\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:305\ngithub.com/matrixorigin/matrixone/pkg/frontend.getDataFromPipeline\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:513\ngithub.com/matrixorigin/matrixone/pkg/sql/colexec/output.Call\n\t/home/mo-221017/matrixone/pkg/sql/colexec/output/output.go:34\ngithub.com/matrixorigin/matrixone/pkg/vm.Run\n\t/home/mo-221017/matrixone/pkg/vm/vm.go:61\ngithub.com/matrixorigin/matrixone/pkg/vm/pipeline.(*Pipeline).MergeRun\n\t/home/mo-221017/matrixone/pkg/vm/pipeline/pipeline.go:147\ngithub.com/matrixorigin/matrixone/pkg/sql/compile.(*Scope).MergeRun\n\t/home/mo-221017/matrixone/pkg/sql/compile/scope.go:110\ngithub.com/matrixorigin/matrixone/pkg/sql/compile.(*Compile).Run\n\t/home/mo-221017/matrixone/pkg/sql/compile/compile.go:101\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2380\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2731\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop\n\t/home/mo-221017/matrixone/pkg/frontend/routine.go:159"}
{"level":"ERROR","time":"2022/10/20 11:46:12.382342 +0800","caller":"frontend/mysql_cmd_executor.go:2600","msg":"internal error: type is not supported in binary text result row","stacktrace":"github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2600\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2731\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop\n\t/home/mo-221017/matrixone/pkg/frontend/routine.go:159"}
{"level":"ERROR","time":"2022/10/20 11:46:12.382419 +0800","caller":"frontend/util.go:465","msg":"query trace status","connection_id":1191,"statement":"execute __mo_stmt_id_63 using @__mo_stmt_var_0","status":"fail","error":"internal error: type is not supported in binary text result row","stacktrace":"github.com/matrixorigin/matrixone/pkg/frontend.logStatementStringStatus\n\t/home/mo-221017/matrixone/pkg/frontend/util.go:465\ngithub.com/matrixorigin/matrixone/pkg/frontend.logStatementStatus\n\t/home/mo-221017/matrixone/pkg/frontend/util.go:457\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).doComQuery\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2608\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest\n\t/home/mo-221017/matrixone/pkg/frontend/mysql_cmd_executor.go:2731\ngithub.com/matrixorigin/matrixone/pkg/frontend.(*Routine).Loop\n\t/home/mo-221017/matrixone/pkg/frontend/routine.go:159"}
```

**Comment by goodMan-code** [comment]
The MO can be inserted with JSON type data,  but can not execute prepare Statement

**Comment by goodMan-code** [comment]
This bug has been fixed, #6566 you can reference this issue, in fact, Mybatis + MO execute prepare to query JSON type data, Successfully NOW. on commit : 873bc541f9ccb4fbb948688a6f6d91ce2c6939c7, let you know.
